config-vault/src/main/java/io/scalecube/config/vault/VaultConfigSource.java

```diff
@@ -27,31 +31,35 @@
   private static final String VAULT_SECRETS_PATH = "VAULT_SECRETS_PATH";
 
   private final VaultInvoker vault;
-  private final String secretsPath;
+  private final List<String> secretsPath;
 
   /**
    * Create a new {@link VaultConfigSource}.
-   *
-   * @param vault vault invoker.
-   * @param secretsPath secret path.
+   *  @param vault vault invoker.
+   * @param secretsPaths secret path-s.
    */
-  private VaultConfigSource(VaultInvoker vault, String secretsPath) {
+  private VaultConfigSource(VaultInvoker vault, List<String> secretsPaths) {
     this.vault = vault;
-    this.secretsPath = secretsPath;
+    this.secretsPath = secretsPaths;
   }
 
   @Override
   public Map<String, ConfigProperty> loadConfig() {
-    try {
-      LogicalResponse response = vault.invoke(vault -> vault.logical().read(secretsPath));
-      return response.getData().entrySet().stream()
-          .map(LoadedConfigProperty::withNameAndValue)
-          .map(LoadedConfigProperty.Builder::build)
-          .collect(Collectors.toMap(LoadedConfigProperty::name, Function.identity()));
-    } catch (Exception ex) {
-      LOGGER.warn("unable to load config properties", ex);
-      throw new ConfigSourceNotAvailableException(ex);
+    Map<String, ConfigProperty> result = new HashMap<>();
+    for (String path : secretsPath) {
+      try {
+        LogicalResponse response = vault.invoke(vault -> vault.logical().read(path));
+        final Map<String, LoadedConfigProperty> pathProps = response.getData().entrySet().stream()
+            .map(LoadedConfigProperty::withNameAndValue)
+            .map(LoadedConfigProperty.Builder::build)
+            .collect(Collectors.toMap(LoadedConfigProperty::name, Function.identity()));
+        result.putAll(pathProps);
+      } catch (Exception ex) {
+        LOGGER.warn("unable to load config properties", ex);
```

Comment:
Specify secret path here

Reply:
thought about it, i think either it was hidden for purpose, or it's part of exception message